### ISSUE TITLE
Normalizer has weird behavior with one-element lists

### ISSUE THREAD

**Issue by TWal** [FStarLang/FStar#3353]
Discussed on Slack, but opening an issue to be able to reference it.

I am noticing a weird behavior with the normalizer, and I wondered what method I should use to understand what is happening?

Here is an example of a proof style I use, that works relatively well:
```fstar
// true when all elements of a list satisfy a predicate
let rec for_allP (#a:Type) (pre:a -> prop) (l:list a): prop =
  match l with
  | [] -> True
  | h::t -> pre h /\ for_allP pre t


// a type, a predicate on that type, and some values
assume val ty: Type0
assume val pre: ty -> prop

assume val v1: ty
assume val v2: ty
assume val v3: ty

// Given a list of values...
let all_v = [ v1; v2; v3 ]

// ... we define this subtle squashed type...
assume val all_v_pre: squash (norm [delta_only [`%all_v; `%for_allP]; iota; zeta] (for_allP pre all_v))

#push-options "--fuel 0 --ifuel 0"
let test () =
  // ... to have `pre ...` in the global scope, with no fuel. Hooray!
  assert(pre v1);
  assert(pre v2);
  assert(pre v3);
  ()
#pop-options
```

However, when the list all_v only contains one element, it is failing to work!

```fstar
let all_v = [ v1 ]

assume val all_v_pre: squash (norm [delta_only [`%all_v; `%for_allP]; iota; zeta] (for_allP pre all_v))

let test () =
  assert(pre v1); // assertion failing :(
  ()
```

@nikswamy found a workaround, inverting the `squash` and `norm`:

```fstar
let all_v = [ v1 ]

assume val all_v_pre: norm [delta_only [`%all_v; `%for_allP]; iota; zeta] (squash (for_allP pre all_v))

#push-options "--fuel 0 --ifuel 0"
let test () =
  assert(pre v1);
  ()
#pop-options
```

**Comment by TWal** [comment]
Some impressive minimization by @mtzguido on Slack:
```fstar
assume val p : Type0
let test (x : squash (p /\ True)) : squash p = ()
```

It reminds me of #2709 and #2694 that also bamboozle F* with some `/\ True`

**Comment by mtzguido** [comment]
Fixed by #3356, thanks!
